plugins/simpleadminhacks-paper/src/main/java/com/programmerdan/minecraft/simpleadminhacks/framework/utilities/BuildLimit.java

```diff
@@ -0,0 +1,50 @@
+package com.programmerdan.minecraft.simpleadminhacks.framework.utilities;
+
+import com.programmerdan.minecraft.simpleadminhacks.configs.buildlimit.LimitType;
+import org.bukkit.configuration.serialization.ConfigurationSerializable;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public record BuildLimit(String world, String type, int minY, int maxY) implements ConfigurationSerializable {
+
+	@NotNull
+	@Override
+	public Map<String, Object> serialize(){
+		Map<String, Object> result = new HashMap<>();
+
+		result.put("world", this.world);
+		result.put("type", this.type);
+		result.put("min_y", this.minY);
+		result.put("max_y", this.maxY);
+
+		return result;
+	}
+	@Nullable
+	public static BuildLimit deserialize(@NotNull final Map<String, Object> args){
+		String world = (String) args.get("world");
+
+        String type = (String) args.get("type");
+        LimitType limitType = LimitType.ALTITUDE;
+        try {
+            limitType = LimitType.valueOf(type);
+        } catch (IllegalArgumentException e) {
+            return null;
```

Comment:
Throwing is fine here as the exception will be handled and printed in a clearer rather than discarded as is what happens here. See `org.bukkit.configuration.serialization.ConfigurationSerialization#deserializeViaMethod`